src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java

```diff
@@ -0,0 +1,264 @@
+
+    servlet.doPost(request, response);
+
+    int actualQueryCount = entitiesInDatastoreCount(LECTURE_ID_C);
+    int expectedQueryCount = (transcriptLines(LONG_VIDEO_JSON)).size();
+    assertThat(actualQueryCount).isEqualTo(expectedQueryCount);
+  }
+
+  @Test
+  public void doGet_onlyOtherLecturesInDatastore_GetNoLectures()
+      throws ServletException, IOException {
+    putTranscriptLinesInDatastore(shortVideoTranscriptLines, LECTURE_ID_B);
+    putTranscriptLinesInDatastore(longVideoTranscriptLines, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_C);
+
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> actualJsonArray = transcriptLines(actualJson);
+    assertThat(actualJsonArray.size()).isEqualTo(0);
+  }
+
+  @Test
+  public void doGet_twoLecturesInDatastore_returnsOneLecture()
+      throws ServletException, IOException {
+    putTranscriptLinesInDatastore(shortVideoTranscriptLines, LECTURE_ID_B);
+    putTranscriptLinesInDatastore(longVideoTranscriptLines, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> expectedArrayList = longVideoTranscriptLines;
+    List<TranscriptLine> actualJsonArray = transcriptLines(actualJson);
+    assertThat(actualJsonArray.size()).isEqualTo(expectedArrayList.size());
+  }
+
+  private static List<TranscriptLine> transcriptLines(String transcriptLinesJson) {
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    return (ArrayList<TranscriptLine>) gson.fromJson(
+      transcriptLinesJson, (new ArrayList<List<TranscriptLine>>().getClass()));
+  }
+
+
+  private void putTranscriptLinesInDatastore(List<TranscriptLine> transcriptLineArray, String lectureKeyId) {
+    Key lectureKeyKey =
+        KeyFactory.createKey(TranscriptServlet.PARAM_LECTURE, Long.parseLong(lectureKeyId));
+    for (int i = 0; i < transcriptLineArray.size(); i++) {
+      Entity lineEntity = new Entity(TranscriptLine.ENTITY_KIND);
+      lineEntity.setProperty(TranscriptLine.PROP_LECTURE, lectureKeyKey);
+      // Set dummy values because AutoValue needs all the values to create a TranscriptLine object.
+      lineEntity.setProperty(TranscriptLine.PROP_START, /* start= */ "");
+      lineEntity.setProperty(TranscriptLine.PROP_DURATION, /* duration= */ "");
+      lineEntity.setProperty(TranscriptLine.PROP_CONTENT, /* content= */ "");
+      datastore.put(lineEntity);
+    }
+  }
+
+  private int entitiesInDatastoreCount(String lectureKeyId) {
+    return datastore.prepare(filteredQuery(lectureKeyId)).countEntities(withLimit(100));
+  }
+
+  private Query filteredQuery(String lectureKeyId) {
```

Comment:
nit: please add a JavaDoc. Or make the method name more specific. I am not certain what filtering you are doing.

Reply:
I updated the method name to be `filteredQueryOfTranscriptLinesByLectureId` Hopefully, that's more clear.